packages/react/src/components/password-creation-input/password-creation.test.tsx

```diff
@@ -0,0 +1,140 @@
+        const event = { target: { name: 'password', value: password } };
+
+        getByTestId(wrapper, 'password-input').prop('onChange')(event);
+
+        expect(onChange).toHaveBeenCalledWith(expect.objectContaining(event), true);
+    });
+
+    test('calls onChange with password and isValid to false when password changes to an invalid password', () => {
+        const wrapper = mountWithProviders(
+            <PasswordCreationInput onChange={onChange} />,
+        );
+        const password = 'Pe23';
+        const event = { target: { name: 'password', value: password } };
+
+        getByTestId(wrapper, 'password-input').prop('onChange')(event);
+
+        expect(onChange).toHaveBeenCalledWith(expect.objectContaining(event), false);
+    });
+
+    test('sets password type to text when show password button is clicked', () => {
+        const wrapper = mountWithProviders(
+            <PasswordCreationInput onChange={onChange} />,
+        );
+
+        getByTestId(wrapper, 'show-password-button').simulate('click');
+
+        expect(getByTestId(wrapper, 'password-input').prop('type')).toBe('text');
+    });
+
+    test('sets password strength to weak when there is only 8 characters', () => {
+        const expectedLabel = Translations.en['password-creation-input']['password-is'] + (
+            Translations.en['password-creation-input'].weak);
+        const wrapper = mountWithProviders(
+            <PasswordCreationInput onChange={onChange} />,
+        );
+
+        getByTestId(wrapper, 'password-input').prop('onChange')({ target: { value: '12345678' } });
+
+        expect(getByTestId(wrapper, 'password-strength').text()).toBe(expectedLabel);
+    });
+
+    test('sets password strength to fair when there is at least 8 characters and at least one uppercase', () => {
+        const expectedLabel = Translations.en['password-creation-input']['password-is'] + (
+            Translations.en['password-creation-input'].fair);
+        const wrapper = mountWithProviders(
+            <PasswordCreationInput onChange={onChange} />,
+        );
+
+        getByTestId(wrapper, 'password-input').prop('onChange')({ target: { value: 'P2345678' } });
+
+        expect(getByTestId(wrapper, 'password-strength').text()).toBe(expectedLabel);
+    });
+
+    test('sets password strength to good when there is at least 8 characters and at least one uppercase', () => {
+        const expectedLabel = Translations.en['password-creation-input']['password-is'] + (
+            Translations.en['password-creation-input'].good);
+        const wrapper = mountWithProviders(
+            <PasswordCreationInput onChange={onChange} />,
+        );
+
+        getByTestId(wrapper, 'password-input').prop('onChange')({ target: { value: 'Pe234567' } });
+
+        expect(getByTestId(wrapper, 'password-strength').text()).toBe(expectedLabel);
+    });
+
+    test('sets password strength to strong when there is at least 8 characters and at least one uppercase', () => {
```

Comment:
Devrait probablement être changé pour:
```suggestion
    test('sets password strength to strong when there is more than 8 characters and at least one lowercase and one uppercase', () => {
```